doc/sg/c_cpp_coding_style.md

```diff
@@ -323,6 +371,13 @@ Macros are often necessary and reasonable coding practice C (as opposed to C++)
 In contrast to the recommendation in the Google C++ style guide, exporting macros as part of the public API is allowed in C code.
 A typical use case is a header with register definitions.
 
+**Function-like macros**
+Function-like macros should be avoided whenever possible since they are error-prone. Where they are necessary, they should be hygienic. Here are some useful tips:
+- Expand the macro arguments between brackets `()` since the caller could use an expression as argument.
+- Variables local to the macro must be named with a trailing underscore `_`.
+- Wrap up multiline macros inside a block `do{...}while(false)` to make them expand to a single statement.
```

Comment:
nit: I would find some spaces more readable: `do { ... } while (false)`.

Would it be worth adding a nice block example? e.g.

```c
#define CHECK(condition, ...)                        \
  do {                                               \
    if (!(condition)) {                              \
      ...                                            \
    }                                                \
  } while (false)
```